client/state/shortcodes/reducer.js

```diff
@@ -46,6 +51,36 @@ export const requesting = createReducer(
 	}
 );
 
+function mediaItemsReducer( state, { siteId, data } ) {
+	if ( ! state.hasOwnProperty( siteId ) ) {
+		return state;
+	}
+
+	if ( ! ( data && data.media ) ) {
+		return state;
+	}
+	const media = castArray( data.media );
```

Comment:
This will not work on a `RECEIVE_MEDIA_ITEM` dispatch. In this case, `data`'s value is the media item and there's no `data.media` property. At least that's how I understand the original Flux code at https://github.com/Automattic/wp-calypso/blob/master/client/lib/shortcodes/store.js#L73-L78

Reply:
nice catch! I totally misread that line.